### ISSUE TITLE
WheelMetadataFinder is searching wrong name

### ISSUE THREAD

**Issue by jaraco** [python/importlib_metadata#22]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 4, 2018, 18:33

Consider:

```
draft $ pip download --no-deps path.py
Collecting path.py
  Using cached https://files.pythonhosted.org/packages/ea/82/d710e0727980ac1dccd2887f866e84c614a68c0a8d0c3aee606997e5eeeb/path.py-11.5.0-py2.py3-none-any.whl
  Saved ./path.py-11.5.0-py2.py3-none-any.whl
Successfully downloaded path.py
draft $ env PYTHONPATH=./path.py-11.5.0-py2.py3-none-any.whl rwt importlib_metadata -- -S
Collecting importlib_metadata
  Using cached https://files.pythonhosted.org/packages/d5/e4/49da554657aebdfe3e26beefc605fabd5dfa054b1864f8a2c527537a3dba/importlib_metadata-0.6-py2.py3-none-any.whl
Installing collected packages: importlib-metadata
Successfully installed importlib-metadata-0.6
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 03:13:28)
[Clang 6.0 (clang-600.0.57)] on darwin
>>> import importlib_metadata
>>> importlib_metadata.distribution('path.py')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-gk1ox5sl/importlib_metadata/api.py", line 86, in distribution
    return Distribution.from_name(package)
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-gk1ox5sl/importlib_metadata/api.py", line 52, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.api.PackageNotFoundError: path.py
```

The metadata isn't found if the distribution name is used. It seems the WheelMetadataFinder is expecting a module name:

```
>>> importlib.distribution('path')
>>> importlib_metadata.distribution('path')
<importlib_metadata._hooks.WheelDistribution object at 0x102ad8780>
```

That's inconsistent with the behavior of the MetadataPathFinder, and violates the user's expectations (that metadata should be readable without importing any modules).

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 10, 2018, 19:33

mentioned in commit ab1bbcf0c9fba2e44784aba40b2575866b523075

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 10, 2018, 23:52

mentioned in commit 1f0d2c72393b04336d537be2e89b4bc39955d656

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 11, 2018, 24:03

mentioned in commit c019aa007a17789f1ceb051265d1135bf0295a6c

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 27, 2018, 17:18

closed via commit c019aa007a17789f1ceb051265d1135bf0295a6c
